lib/matplotlib/axes/_axes.py

```diff
@@ -5164,20 +5187,32 @@ def fill_between(self, x, y1, y2=0, where=None, interpolate=False,
             where = True
         else:
             where = np.asarray(where, dtype=bool)
-            if where.size != x.size:
+            if where.size != x.size and step != 'between':
                 cbook.warn_deprecated(
                     "3.2",
                     message="The parameter where must have the same size as x "
                             "in fill_between(). This will become an error in "
                             "future versions of Matplotlib.")
-        where = where & ~functools.reduce(np.logical_or,
-                                          map(np.ma.getmask, [x, y1, y2]))
 
-        x, y1, y2 = np.broadcast_arrays(np.atleast_1d(x), y1, y2)
+        if step == 'between':
+            pad_size = x.size - 1
+        else:
+            pad_size = x.size
+        y1 = np.broadcast_to(y1, pad_size, subok=True)
+        y2 = np.broadcast_to(y2, pad_size, subok=True)
+        where = np.broadcast_to(where, pad_size, subok=True)
+
+        get_masks = cbook.pad_arrays(list(map(np.atleast_1d,
```

Comment:
I actually don't understand this `pad_arrays` usage here.  Is there some idea that the masks will be different lengths?  How is that possible?

Reply:
Cool, I didn't notice I made this obsolete by the lines above.